modules/sfp_flickr.py

```diff
@@ -100,67 +130,101 @@ def handleEvent(self, event):
 
         self.sf.debug("Retrieved API key: " + api_key)
 
-        # Get e-mail addresses for this domain
+        # Query API for event data
+        hosts = list()
         page = 1
         pages = self.opts['maxpages']
         per_page = self.opts['per_page']
         while page <= pages:
             if self.checkForStop():
                 return None
 
-            res = self.query("@"+eventData, api_key, page=page, per_page=per_page)
-
-            if res is None:
+            if self.errorState:
                 return None
 
-            # Parse response content as JSON
-            try:
-                data = json.loads(res)
-            except Exception as e:
-                self.sf.debug("Error processing JSON response.")
+            data = self.query(eventData, api_key, page=page, per_page=per_page)
+
+            if data is None:
                 return None
 
             # Check the response is ok
-            if not data['stat'] == "ok":
+            if data.get('stat') != "ok":
                 self.sf.debug("Error retrieving search results.")
                 return None
 
-            if 'photos' not in data:
-                return None
+            photos = data.get('photos')
 
-            if 'pages' not in data['photos']:
+            if not photos:
                 return None
 
             # Calculate number of pages to retrieve
-            result_pages = int(data['photos']['pages'])
+            result_pages = int(photos.get('pages'))
 
             if result_pages < pages:
                 pages = result_pages
 
-            if 'max_allowed_pages' in data['photos']:
-                allowed_pages = int(data['photos']['max_allowed_pages'])
+            if 'max_allowed_pages' in photos:
+                allowed_pages = int(photos.get('max_allowed_pages'))
                 if pages > allowed_pages:
                     pages = allowed_pages
 
             self.sf.info("Parsing page " + str(page) + " of " + str(pages))
 
-            # Extract emails
-            for photo in data['photos']['photo']:
-                emails = self.sf.parseEmails(str(bytes(photo).decode('unicode-escape')))
+            # Extract data
+            for photo in photos.get('photo'):
```

Comment:
As above, there's a chance this could be `None`, so expecting to iterate it would raise an exception. Maybe set the default return value to `list()` with `get()` to be on the safe side?